st2reactor/st2reactor/rules/filter.py

```diff
@@ -149,7 +157,38 @@ def _render_criteria_pattern(self, criteria_pattern):
             # makes no sense
             return criteria_pattern
 
-        criteria_pattern = render_template_with_system_context(value=criteria_pattern)
+        LOG.debug(
+            'Rendering criteria pattern (%s) with context: %s',
+            criteria_pattern,
+            criteria_context
+        )
+
+        to_complex = False
+
+        # Check if jinja variable is in criteria_pattern and if so lets ensure
+        # the proper type is applied to it using to_complex jinja filter
+        if len(re.findall(r'^{{\s*[A-z0-9_.-]+\s*}}$', criteria_pattern)) > 0:
+            LOG.debug("Rendering Complex")
+            criteria_pattern = re.sub(
+                r'^{{\s*([A-z0-9_.-]+)\s*}}$', r'{{\1 | to_complex}}',
```

Comment:
It will match `{{ test.var }}` and `{{ one_var.two_var }}`. It will not match `{{ variable | my_filter }}` or anything more complex. It also won't match `pre{{ variable }}post`